lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -1976,10 +1976,20 @@ SwiftExpressionParser::Parse(DiagnosticManager &diagnostic_manager,
 
   // If IRGen failed without errors, the root cause may be a fatal
   // Clang diagnostic.
-  if (expr_diagnostics->HasErrors() ||
-      m_swift_ast_ctx.HasClangImporterErrors()) {
-    diagnostic_manager.PutString(eDiagnosticSeverityRemark,
-                                 "couldn't IRGen expression");
+  using ErrorKind = SwiftASTContext::ScopedDiagnostics::ErrorKind;
+  std::optional<ErrorKind> error_kind = expr_diagnostics->GetErrorKindIfAny();
+  if (error_kind || m_swift_ast_ctx.HasClangImporterErrors()) {
```

Comment:
What would lead to `error_kind` being null, and yet `HasClangImporterErrors` being true?

Reply:
See the comment I suggested above :-)

Reply:
oops! yes good comment